src/util/expandThoughts.js

```diff
@@ -80,31 +82,36 @@ export const expandThoughts = (path, thoughtIndex, contextIndex, contextViews =
   return (isOnlyChildNoUrl || isTable() || pinChildren() || publishPinChildren
     ? children
     : children.filter(child => {
+      const childPath = (path || []).concat(child)
+      const isCursor = equalPath(childPath, cursor)
+      const isDirectAncestorOfCursor = isDescendant(childPath, cursor)
```

Comment:
This is an expensive operation within a loop. It seems like we've traded extra recursion for extra computation. I'm not really satisfied with this solution.